ArduCopter/flight_mode.cpp

```diff
@@ -26,6 +26,11 @@ bool Copter::set_mode(control_mode_t mode, mode_reason_t reason)
         return true;
     }
 
+    // check if an RTL needs to be converted to an AUTO
+    if (mode == RTL && g2.rtl_type == RTLType_MissionDoLandStart) {
+        mode = check_do_land_start();
+    }
+    
```

Comment:
we do actually need to run auto_init() in this case, even if doing AUTO -> AUTO, we need the mission.start_or_resume()